django/contrib/postgres/forms/array.py

```diff
@@ -216,3 +216,23 @@ def clean(self, value):
         if errors:
             raise ValidationError(list(chain.from_iterable(errors)))
         return cleaned_data
+
+    def has_changed(self, initial, data):
+        try:
+            python_data = self.to_python(data)
+        except ValidationError:
+            pass
+        else:
+            if self.remove_trailing_nulls:
+                null_index = None
+                for i, value in reversed(list(enumerate(python_data))):
```

Comment:
Yep, you are right.

Reply:
Actually, it is required - `TypeError: 'enumerate' object is not reversible`. So I kept it there.

Reply:
That actually makes sense. I forgot `reversed()` is a bit special because you typically have to have consumed the whole iterator to know the last element.